web/themes/contrib/civictheme/includes/views.inc

```diff
@@ -261,3 +276,83 @@ function _civictheme_preprocess_views_view__search_page(array &$variables): void
     $variables['vertical_spacing'] = 'top';
   }
 }
+
+/**
+ * Get an array of the active filters and format them for the selected filters.
+ */
+function _civictheme_automated_list__selected_filters(): array {
+  $query_params = \Drupal::request()->query->all();
+  $current_values = array_filter($query_params, function ($value) {
+    return !empty($value);
+  });
+
+  // Prevent other query params from being used in the selected filters.
+  $permitted_keys = [
+    'type',
+    'topic',
+    'title',
+    'items_per_page',
+  ]; // Change this to get from $view there is an exposed input field or something that gives allowed exposed filters, pass $view in as argument to this function
+  // this function should be called after the hook for changing view
+    $selected_filters = [];
+  foreach ($current_values as $key => $value) {
+    if (!in_array($key, $permitted_keys)) {
+      continue;
+    }
+    $new_query_params = $query_params;
+    if (is_string($value)) {
+      unset($new_query_params[$key]);
+      $selected_filters[$key] = [
+        'url' => \Drupal::request()->getUriForPath(\Drupal::request()->getPathInfo()) . '?' . http_build_query($new_query_params),
+        'text' => _civictheme_automated_list__create_filter_label($key, $value),
+      ];
+      continue;
+    }
+    if (is_array($value)) {
+      foreach ($value as $value_key => $item) {
+        if (!empty($item)) {
+          $temp_query_params = $query_params;
+          unset($temp_query_params[$key][$value_key]);
+          if (empty($temp_query_params[$key])) {
+            unset($temp_query_params[$key]);
+          }
+          $selected_filters[$key . '_' . $item . '_' . $value_key] = [
+            'url' => \Drupal::request()->getUriForPath(\Drupal::request()->getPathInfo()) . '?' . http_build_query($temp_query_params),
+            'text' => _civictheme_automated_list__create_filter_label($key, $item),
+          ];
+        }
+      }
+    }
+  }
+  return $selected_filters;
+}
+
+/**
+ * Create the label correctly for each of the selected filters.
+ *
+ * @param string $key
+ *   The filter key.
+ * @param string|int $value
+ *   The filter value.
+ *
+ * @return string
+ *   The formatted filter label.
+ */
+function _civictheme_automated_list__create_filter_label(string $key, string|int $value): string {
+  // Get the value label.
+  switch ($key) {
+    case 'type':
```

Comment:
Is there a more generic way we can check what kind of filter this is and apply the necessary case. E.g. if a filter based on nodes, then load the node to get label, if a filter based on taxonomy then load taxonomy term label.

Reply:
Yeah can we use the view object to find out what they are filters of?

Reply:
Thanks @alan-cole, 
I tried but couldn't implemented a more generic way, Could you please support me with this.
Thank you!

Reply:
I've implemented a change to use `bundle` and `taxonomy_index_tid` types to determine how to get the label, rather than hard-coding the `type` and `topic` keys in.